### ISSUE TITLE
torch.optim.lr_scheduler.ReduceLROnPlateau not registered

### ISSUE THREAD

**Issue by lgessler** [allenai/tango#482]
### 🐛 Describe the bug

For some reason, PyTorch's `ReduceLROnPlateau` learning rate scheduler is [not registered as a subclass of `LRScheduler`](https://github.com/pytorch/pytorch/blob/564905c8e1dbd08a773b48e81e3413883e99ca7e/torch/optim/lr_scheduler.py#L913). This causes [Tango's registration to fail to register it](https://github.com/allenai/tango/blob/4a183de94fefb6bb8027f0797ce1e77be4083f6d/tango/integrations/torch/optim.py#L76L83), as Tango uses a subclass of `LRScheduler` check to see whether a given class ought to be registered.

Suggested fix: manually include `ReduceLROnPlateau` in Tango's registration code. I'd be happy to open a PR for this.

### Versions

```
ai2tango==1.0.2
torch==1.13.0
```

**Comment by lgessler** [comment]
OK, I actually see that `ReduceLROnPlateau.step` actually requires a `metrics` argument, which would complicate Tango's training implementation, so the fix is not as easy as I thought.
